core/cap-0038.md

```diff
@@ -218,26 +218,21 @@ index 0e7bc842b..2976045dc 100644
 +    int32 fee;    // Fee is in basis points, so the actual rate is (fee/100)%
 +};
 +
-+union LiquidityPoolEntry switch (LiquidityPoolType type)
++struct LiquidityPoolEntry
```

Comment:
Probably want to change the name of this now to something more specific. Perhaps `LiquidityPoolConstantProductEntry`?

Reply:
Yeah I was thinking about that. I can make the change if @MonsieurNicolas thinks this PR as a whole makes sense.

Reply:
Maybe?

This will cause code to pass around `LedgerEntry` instead of `LiquidityPoolEntry`, so this makes sharing code only to be used for pools a little more clunky (as they have to check with pre-conditions that a pool was passed in).
Basically we'd be using naming conventions to enforce certain things, which is not strong and potentially error prone.

Reply:
So this is something that came up when @jonjove and I discussed this issue. The conclusion we came to was that the switch on the union would just happen one level lower, so there isn't much more risk. I'm fine with changing this to use a union though if you think that's the right thing to do and Jon doesn't have any issues with that approach.

Reply:
@MonsieurNicolas What do you think about our reasoning for this? I’d like to close on this soon so I can finish up with the pool entry schema change.

Reply:
I think I prefer what is in master:
* master allows to reference "pool related types" in a strict way (ie: enforced by the compiler), in core but also in any other code and avoids using naming conventions. This seems better in terms of future proofing code/debuggability.
* the proposed change is not that much simpler, and seems like a wash in the context of CAP38 implementation.

Reply:
Ok I think using a union in `LiquidityPoolEntry` is fine. 

Now we either need to move `liquidityPoolID` out of that union, or use a union for the `LIQUIDITY_POOL` `LedgerKey`. I would prefer to pull `liquidityPoolID` out of the union since I expect every pool to use a hash for a key, and adding a union to the `LIQUIDITY_POOL` `LedgerKey` adds unnecessary complexity. What do you think @jonjove?

Reply:
I'm fine with pulling the hash out of the union. I can't imagine a case where we wouldn't use a hash as the key, and even if there are cases where we would prefer something simpler the hash would still be an acceptable choice. Plus if we really needed to do something different we could always create a new type of `LedgerEntry` anyway.